sms_sponsorship/forms/sms_sponsorship_registration_form.py

```diff
@@ -186,11 +188,22 @@ def form_before_create_or_update(self, values, extra_values):
                     })
             sponsorship.on_change_group_id()
 
-        def form_after_create_or_update(self, values, extra_values):
+        @job(default_channel="root.res_partner")
+        @related_action(action='related_action_update_partner')
+        def update_partner(self, partner_vals):
+            self.env['res.partner'].search([
```

Comment:
If you put this method inside res.partner object, you can simply do `return self.write(partner_vals)`